### PR TITLE
Hibernate

### PR DESCRIPTION
**Pull request by Borisina**

### CODE REVIEW

File: logs/2023-12/app-12-29-2023-1.log.gz
Comment (FloatingPirate) [inline logs/2023-12/app-12-29-2023-1.log.gz:1]: Why do we need log files in the repository. 
Let's ignore them.

File: src/main/java/com/kolya/gym/Main.java [hunk 1]
```diff
@@ -28,26 +28,35 @@ public static void main(String[] args) {
...
         traineeData3.setLastName("Alexeev");
-        traineeFacade.createTrainee(traineeData3);
+        Trainee trainee= traineeFacade.createTrainee(traineeData3);
+        AuthData authData = new AuthData(trainee.getUser().getUsername(),trainee.getUser().getPassword());
+
+        TraineeData traineeData4 = new TraineeData();
+        traineeData4.setFirstName("Vasiliy");
+        traineeData4.setLastName("Palkin");
+        traineeFacade.createTrainee(traineeData4);
+
```
Comment (FloatingPirate) [inline src/main/java/com/kolya/gym/Main.java:40]: Let's create training and trainee data by using "builder" pattern in a separate file.  In order to unclutter "main" method.

File: src/main/java/com/kolya/gym/validation/CommonValidation.java [hunk 2]
```diff
@@ -0,0 +1,10 @@
+package com.kolya.gym.validation;
+
+public class CommonValidation {
+    static public void nullValidation(Object object) throws IllegalArgumentException{
+        if (object==null){
+            throw new IllegalArgumentException("Input Data is null");
```
Comment (FloatingPirate) [inline src/main/java/com/kolya/gym/validation/CommonValidation.java]: Better to use `Optional` class instead of a separate class for null verification.

**Review by FloatingPirate — APPROVED** [review]

File: src/main/java/com/kolya/gym/validation/CommonValidation.java (hunk 2 above)
Comment (Borisina) [reply src/main/java/com/kolya/gym/validation/CommonValidation.java]: I replaced all "nullValidation()" functions with "Optional.ofNullable().OrElseThrow()"

File: logs/2023-12/app-12-29-2023-1.log.gz
Comment (Borisina) [reply logs/2023-12/app-12-29-2023-1.log.gz:1]: Logs ignored and deleted from git.

File: src/main/java/com/kolya/gym/Main.java (hunk 1 above)
Comment (Borisina) [reply src/main/java/com/kolya/gym/Main.java:40]: Now training, trainee and trainer datas are created in the separated class "PreparedData" by using builders.

**Pull request merged**